posthog/models/cdp/filters.py

```diff
@@ -0,0 +1,102 @@
+from typing import Optional
+
+from typing import Any
+from posthog.models.action.action import Action
+from posthog.hogql.bytecode import create_bytecode
+from posthog.hogql.parser import parse_expr, parse_string_template
+from posthog.hogql.property import action_to_expr, property_to_expr, ast
+from posthog.models.team.team import Team
+
+
+# Helper methods for CDP filters (used by HogFunctions and Batch Exports)
+# Provides methods for compiling filters into bytecode in a standard format
+
+
+def compile_filters_bytecode(team: Team, filters: dict, actions: Optional[dict[int, Action]] = None) -> dict:
+    """
+    Re-compiles filters into bytecode format. This should be called whenever filters, actions or the team test account filters change.
+    """
+    filters = filters or {}
+
+    if actions is None:
+        filter_action_ids = get_action_ids_in_filters(filters)
+        # If not provided as an optimization we fetch all actions
+        actions_list = Action.objects.select_related("team").filter(team_id=team.id).filter(id__in=filter_action_ids)
+        actions = {action.id: action for action in actions_list}
+
+    try:
+        filters["bytecode"] = create_bytecode(hog_function_filters_to_expr(filters, team, actions))
+    except Exception as e:
+        filters["bytecode"] = None
+        filters["bytecode_error"] = str(e)
+
+    return filters
+
+
+def get_action_ids_in_filters(filters: dict) -> list[int]:
+    """
+    Extracts action IDs from filters
+    """
+    filters = filters or {}
+
+    return [int(action["id"]) for action in filters.get("actions", [])]
+
+
+def hog_function_filters_to_expr(filters: dict, team: Team, actions: dict[int, Action]) -> ast.Expr:
```

Comment:
It's kind of hard to understand the resulting expression from reading the function. Are there any examples we could look at? Potentially in a test case to ensure this thing works?

Reply:
I am mostly interested in the sql an action filter transpiles to, as I have concerns about performance.

Reply:
My knowledge here breaks down but maybe @mariusandra can be of help? 
It populates to the same expression that a HogQL expression would be, but I'm not sure what the correct way of then using it in a clickhouse query is tbh

Reply:
It may be as well that the whole code there doesn't apply for batch exports so we can definitely un-abstract it if thats the case

Reply:
Yeah, I did have some follow-up questions regarding how actions would interact with batch exports:
* What if some of the events that compose the action fall in a previous batch? Should that action be included in the current batch? I guess these questions fall under "When does an action happen?".
* Should we extend the schema with an `action_id`?
* General concerns about performance giving that filter fields may not be in the sort key, and sampling is not a valid strategy for batch exports.

I hoped seeing the SQL would help me clarify these. In the meantime though, it may be easier to limit to event filters only for batch exports, as we already work with those, and they are part of the sort key.

Reply:
As far as I understand, actions are just events and properties anyways (other than cohorts but those will not cause a compile error so also don't apply.
If you think of an "Action" as a "Saved Filter" then it is easier to reason on in that I would say it should simply filter the data being exported